### PR TITLE
Rewrite player

### LINKED ISSUE #305: Automatically start playing upon requesting a song
When Spotify is not playing, requesting a song should also start the playback

### LINKED ISSUE #409: Fix player_start bug without active device
Every morning, we see this:

```
Traceback (most recent call last):
  File "/var/www1/tosti/live/repo/env/lib/python3.8/site-packages/spotipy/client.py", line 245, in _internal_call
    response.raise_for_status()
  File "/var/www1/tosti/live/repo/env/lib/python3.8/site-packages/requests/models.py", line 1021, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 404 Client Error: Not Found for url: https://api.spotify.com/v1/me/player/play

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/var/www1/tosti/live/repo/env/lib/python3.8/site-packages/django_cron/management/commands/runcrons.py", line 82, in run_cron_with_cache_check
    manager.run(force)
  File "/var/www1/tosti/live/repo/env/lib/python3.8/site-packages/django_cron/__init__.py", line 282, in run
    self.msg = self.cron_job.do()
  File "/var/www1/tosti/live/repo/website/thaliedje/crons.py", line 32, in do
    player.spotify.start_playback()
  File "/var/www1/tosti/live/repo/env/lib/python3.8/site-packages/spotipy/client.py", line 1776, in start_playback
    return self._put(
  File "/var/www1/tosti/live/repo/env/lib/python3.8/site-packages/spotipy/client.py", line 312, in _put
    return self._internal_call("PUT", url, payload, kwargs)
  File "/var/www1/tosti/live/repo/env/lib/python3.8/site-packages/spotipy/client.py", line 267, in _internal_call
    raise SpotifyException(
spotipy.exceptions.SpotifyException: http status: 404, code:-1 - https://api.spotify.com/v1/me/player/play:
 Player command failed: No active device found, reason: NO_ACTIVE_DEVICE

```

Probably adding the device_uri to the start_playback method solves it (but no guarantees)

### PR DESCRIPTION
**Pull request by JobDoesburg**
Closes #305 
Works on #254 which is now waiting for marietjee
Hopefully fixes #409

### CODE REVIEW

File: website/thaliedje/views.py [hunk 1]
```diff
@@ -33,31 +32,40 @@ def get_context_data(self, **kwargs):
...
-            context["current_control_event"] = control_event
-
-        if self.request.user.is_authenticated:
-            context["can_request_song"] = can_request_song(self.request.user, player)
-            context["can_request_playlist"] = can_request_playlist(self.request.user, player)
-            context["can_control_player"] = can_control_player(self.request.user, player)
-        else:
+        try:
+            player = SpotifyPlayer.objects.get(pk=player.pk)
+        except SpotifyPlayer.DoesNotExist:
```
Comment (KiOui) [inline website/thaliedje/views.py:39]: Shouldn't this also be abstract when we are using an abstract player model?

File: website/thaliedje/migrations/0010_marietjeplayer_remove_player_client_id_and_more.py [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+from django.conf import settings
+from django.db import migrations, models
+import django.db.models.deletion
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        migrations.swappable_dependency(settings.AUTH_USER_MODEL),
+        ("thaliedje", "0009_playerlogentry_thaliedjecontrolevent"),
```
Comment (KiOui) [inline website/thaliedje/migrations/0010_marietjeplayer_remove_player_client_id_and_more.py:12]: This migration does not migrate old Spotify clients to new ones.

File: website/thaliedje/converters.py [hunk 3]
```diff
@@ -13,7 +13,11 @@ def to_python(self, value):
         :return: a Player or ValueError
         """
         try:
-            return Player.objects.get(slug=value)
+            player = Player.objects.get(slug=value)
+            try:
+                return player.spotifyplayer
+            except SpotifyPlayer.DoesNotExist:
+                return player
```
Comment (KiOui) [inline website/thaliedje/converters.py:20]:
```suggestion
            except SpotifyPlayer.DoesNotExist:
                return player.marietjeplayer
```

**Review by KiOui — CHANGES_REQUESTED** [review]

**Review by KiOui — APPROVED** [review]

File: website/thaliedje/migrations/0010_marietjeplayer_remove_player_client_id_and_more.py (hunk 2 above)
Comment (JobDoesburg) [reply website/thaliedje/migrations/0010_marietjeplayer_remove_player_client_id_and_more.py:12]: Can't be fixed in a nice manner

**Pull request merged**